### ISSUE TITLE
Virtual Areas

### ISSUE THREAD

**Issue by User873902** [agittins/bermuda#180]
It would seem possible to create virtual areas to represent rooms without ESPHome base stations. Observed distances from each base station would localize to a point in the center of the virtual area. A radius or range of distances would define the room. 

Is this a planned or available feature?

Person = n
Base station = A, B, C, D, E
Distance = d
dAn = distance between base station (A) and person (n)
dAn1 = distance between base station (A) and person (n) in the desired room, closest to base station (A)
dAn2 = distance between base station (A) and person (n) in the desired room, furthest from the base station (A)
Virtual location = (dAn, dBn, dCn, dDn)
Virtual Area/Room = (dAn1 < dAn <dAn2, dBn1 < dBn < dBn2, dCn1 < dCn < dCn2, dDn1 < dDn < dDn2)

**Comment by agittins** [comment]
Yes, this is definitely planned once we have trilateration working. At this stage I'm expecting to generate x/y(maybe z) co-ords that map to the floorplan.

I haven't quite settled on how we'd translate that to rooms directly, i think we'd define polygons for each room, and some magical numpy or pandas library would take that and tell you which room any given set of co-ords falls in.

**Comment by nebulous** [comment]
I was here to suggest this very thing.  (_Pats self on back for searching existing issues first_)

I know the more ambitious goal of Bermuda is to do fancy math so solve for the entire set/general case, but based on my limited measurements from proxies in my house, it looks like virtual areas can also be solved using more brutish configuration. Especially when multiple floors are concerned.

Contrived example:
```mermaid
flowchart LR
    lr((LivingRoom)) --- Hallway --- dr((DiningRoom))
```

If I had a proxy in both LivingRoom and DiningRoom, then there would be some threshold combination of values which indicated `hallway` vs either room. Instead of polygon definitions I'd be happy enough with defining a set of spherical distances from each proxy with a "thickness"/error band for each or even just a Jinja blob to set an area based on arbitrary inputs.

In my actual use case with a multi-story house, when I walk from my office to the kitchen I walk over the kids playroom in the basement. It's impossible* to walk through the basement playroom in between the office and kitchen, but an intersection of `officeproxy` and `kitchenproxy` and `playroomproxy` *is* the dining room.


*Could overcomplicate this further by enforcing some state transition validations.
